source/guides/guides/parallelization.md

```diff
@@ -92,10 +121,127 @@ Machine 1 (Total of 39s)                  Machine 2 (Total of 38s)
+
+```shell
+cypress run --record --group 2x-chrome --browser chrome
+```
+
+```shell
+cypress run --record --group 4x-electron
+```
+
+The `1x`, `2x`, `4x` group prefix used here is simply an adopted convention to indicate the level of parallelism for each run, and *is not required or essential*.
+
+{% note info %}
+Please do take into account the number machines dedicated for each `cypress run` call is based on your CI configuration for the project.
+{% endnote %}
+
+These grouped runs would look something like this within the Cypress Dashboard:
+
+{% img /img/guides/parallelization/timeline-collapsed.png "Timeline view with grouping and parallelization" %}
+
+## Grouping by spec context
+
+Lets say you have an application that has a *customer facing portal* and an *administration facing portal*. You could organize and test these two parts of your application within the same run:
+
+- One group can be called `customer-portal`:
+
+  ```shell
+  cypress run --record --group customer-portal --spec 'cypress/integration/portals/customer/**/*'
+  ```
+
+- The other group can be called `admin-portal`:
+
+  ```shell
+  cypress run --record --group admin-portal --spec 'cypress/integration/portals/admin/**/*'
+  ```
+
+This pattern is especially useful for monorepo projects. Each segment of the monorepo can be assigned its own group, and larger segments could be parallelized to speed up their testing.
+
+# Linking CI machines for parallelization or grouping
+
+A CI build ID is used to associate multiple CI machines to a parallelized or grouped test run. This identifier is based on environment variables that are unique to each CI build, and vary based on CI provider. Cypress has out-of-the-box support for most of the commonly-used CI providers, so you would typically not need to directly set the CI build ID via the {% url "`--ci-build-id` flag" command-line#cypress-run-ci-build-id-lt-id-gt %} as Cypress automatically handles this for you.
+
+{% img 'no-border' /img/guides/parallelization/ci-build-id.png "CI Machines linked by ci-build-id" %}
+
+## CI Build ID environment variables by provider
+
+Cypress currently uses the following CI environment variables to determine a CI build ID for a test run:
+
+| Provider  | Environment Variable  |
+|--|--|
+| Appveyor  | `APPVEYOR_BUILD_NUMBER`  |
+| Bamboo  | `BAMBOO_BUILD_NUMBER`  |
+| Circle  |  `CIRCLE_WORKFLOW_ID`, `CIRCLE_BUILD_NUMBER` |
+| Codeship  | `CI_BUILD_NUMBER`  |
+| Codeship Basic  | `CI_BUILD_NUMBER`  |
+| Codeship Pro  | `CI_BUILD_ID`  |
+| Drone  | `DRONE_BUILD_NUMBER`  |
+| Gitlab  | `CI_PIPELINE_ID`, `CI_JOB_ID`, `CI_BUILD_ID`  |
+| Jenkins  | `BUILD_NUMBER`  |
+| Travis  | `TRAVIS_BUILD_ID`  |
+
+# Run completion delay
+
+During parallelization mode or when grouping runs, Cypress will wait for a specified amount of time before closing down and completing the test run in case any more relevant work. This is to compensate for various scenarios where CI machines could be backed-up in a queue.
```

Comment:
Maybe "in case any more relevant work remains"

Reply:
Might be worth adding a scenario of when you might want to change this value somewhere in this chunk. "If you find your tests are <whatever>, you could up this value to 90." or something

Reply:
@lilaconlee good idea